src/promql/src/planner.rs

```diff
@@ -1303,6 +1264,93 @@ impl PromPlanner {
         Ok(exprs)
     }
 
+    /// Create a [SPECIAL_HISTOGRAM_QUANTILE] plan.
+    async fn create_histogram_plan(&mut self, args: &PromFunctionArgs) -> Result<LogicalPlan> {
+        if args.args.len() != 2 {
+            return FunctionInvalidArgumentSnafu {
+                fn_name: SPECIAL_HISTOGRAM_QUANTILE.to_string(),
+            }
+            .fail();
+        }
+        let phi = Self::try_build_float_literal(&args.args[0]).with_context(|| {
+            FunctionInvalidArgumentSnafu {
+                fn_name: SPECIAL_HISTOGRAM_QUANTILE.to_string(),
+            }
+        })?;
+        let input = args.args[1].as_ref().clone();
+        let input_plan = self.prom_expr_to_plan(input).await?;
+
+        if !self.ctx.has_le_tag() {
+            return ColumnNotFoundSnafu {
+                col: LE_COLUMN_NAME.to_string(),
+            }
+            .fail();
+        }
+        let time_index_column =
+            self.ctx
+                .time_index_column
+                .clone()
+                .with_context(|| TimeIndexNotFoundSnafu {
+                    table: self.ctx.table_name.clone().unwrap_or_default(),
+                })?;
+        // FIXME(ruihang): support multi fields
```

Comment:
I think this feature should be raised the priority.

Reply:
This `FIXME` is not introduced in this PR. And unlike other functions, `histogram_quantile` has a more strict requirement on the field column. Actually I'm doubting if it's recommended to use this function on multi-field tables -- prometheus protocol only generates one-field table, and other trivial, user-created tables can hardly meet the requirement of `histogram_quantile` function.

Reply:
I meant the total multi-field feature, not just `vector` or `histogram_quantile`. Not sure about the current status.

Reply:
I'm not sure what you refer to. Run queries on multi-field tables is already supported since a very early stage.

Reply:
I meant if the multi-field feature is supported by all the functions that can support it.

Reply:
So we are talking about `vector` and `histogram_quantile`. If we define them as "unable" then this feature
>is supported by all the functions that are able to support it.